Lesson25/Lesson25-fercan/filter.js

```diff
@@ -0,0 +1,429 @@
+	},
+	{
+		name: 'Smart Speaker',
+		brand: 'Amazon',
+		price: 99,
+		quantity: 0,
+	},
+	{
+		name: 'Fitness Tracker',
+		brand: 'Fitbit',
+		price: 129,
+		quantity: 14,
+	},
+];
+
+const producstInStock = productsExercise6.filter(object => object.quantity>0);
+console.log('Ex.6:');
+console.log(producstInStock);
+
+// Exercise 7:
+// Given an array of numbers, filter out the numbers that are divisible by 3.
+const numbersExercise7 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+const filteredNumbers = numbersExercise7.filter(number => number % 3 !== 0);
+console.log('Ex.7:');
+console.log(filteredNumbers);
+
+// Exercise 8:
+// Given an array of strings, filter out the strings that have more than 5 characters.
+const wordsExercise8 = [
+	'apple',
+	'banana',
+	'car',
+	'dog',
+	'elephant',
+	'flower',
+	'guitar',
+	'house',
+	'island',
+	'jungle',
+];
+console.log('Ex.8:');
+const wordsLessThan5 = wordsExercise8.filter(word => word.length<=5);
+console.log(filteredWords);
+
+
+// Exercise 9:
+// Given an array of objects representing students, filter out the students who have a grade lower than C.
+const studentsExercise9 = [
+	{
+		id: 1,
+		name: 'Alice',
+		age: 18,
+		grade: 'A',
+	},
+	{
+		id: 2,
+		name: 'Bob',
+		age: 19,
+		grade: 'B',
+	},
+	{
+		id: 3,
+		name: 'Charlie',
+		age: 20,
+		grade: 'B+',
+	},
+	{
+		id: 4,
+		name: 'David',
+		age: 19,
+		grade: 'A-',
+	},
+	{
+		id: 5,
+		name: 'Emma',
+		age: 18,
+		grade: 'A+',
+	},
+	{
+		id: 6,
+		name: 'Frank',
+		age: 20,
+		grade: 'B',
+	},
+	{
+		id: 7,
+		name: 'Grace',
+		age: 19,
+		grade: 'D+',
+	},
+	{
+		id: 8,
+		name: 'Hannah',
+		age: 20,
+		grade: 'A-',
+	},
+	{
+		id: 9,
+		name: 'Isaac',
+		age: 18,
+		grade: 'D',
+	},
+	{
+		id: 10,
+		name: 'Julia',
+		age: 19,
+		grade: 'B-',
+	},
+];
+
+console.log('Ex.9:');
+
+const studentsStudyMore = studentsExercise9.filter(student => student.grade<'C');
+console.log(studentsStudyMore);
+
+
+// Exercise 10:
+// Given an array of user objects, filter out the users who have a name starting with the letter "J".
+const usersExercise10 = [
+	{
+		id: 1,
+		name: 'John',
+		age: 25,
+		email: 'john@example.com',
+	},
+	{
+		id: 2,
+		name: 'Alice',
+		age: 32,
+		email: 'alice@example.com',
+	},
+	{
+		id: 3,
+		name: 'Bob',
+		age: 28,
+		email: 'bob@example.com',
+	},
+	{
+		id: 4,
+		name: 'Emily',
+		age: 21,
+		email: 'emily@example.com',
+	},
+	{
+		id: 5,
+		name: 'Michael',
+		age: 35,
+		email: 'michael@example.com',
+	},
+	{
+		id: 6,
+		name: 'Sara',
+		age: 29,
+		email: 'sara@example.com',
+	},
+	{
+		id: 7,
+		name: 'David',
+		age: 42,
+		email: 'david@example.com',
+	},
+	{
+		id: 8,
+		name: 'Olivia',
+		age: 31,
+		email: 'olivia@example.com',
+	},
+	{
+		id: 9,
+		name: 'Daniel',
+		age: 27,
+		email: 'daniel@example.com',
+	},
+	{
+		id: 10,
+		name: 'Sophia',
+		age: 23,
+		email: 'sophia@example.com',
+	},
+];
+console.log('Ex.10:');
+const filteredUsers = usersExercise10.filter(user => !user.name.startsWith('J'));
```

Comment:
It would be a safer approach to convert the first letter to capital letter as you compare it to a capital letter ('j' is not equal to 'J' in JS).
```suggestion
const filteredUsers = usersExercise10.filter(user => !user.name.toUpperCase().startsWith('J'));
```